session/tidb.go

```diff
@@ -165,12 +165,23 @@ func Compile(ctx context.Context, sctx sessionctx.Context, stmtNode ast.StmtNode
 func finishStmt(ctx context.Context, sctx sessionctx.Context, se *session, sessVars *variable.SessionVars,
 	meetsErr error, sql sqlexec.Statement) error {
 	if meetsErr != nil {
+		var aborted bool
 		if !sessVars.InTxn() {
 			logutil.BgLogger().Info("rollbackTxn for ddl/autocommit failed")
 			se.RollbackTxn(ctx)
+			aborted = true
 		} else if se.txn.Valid() && se.txn.IsPessimistic() && executor.ErrDeadlock.Equal(meetsErr) {
 			logutil.BgLogger().Info("rollbackTxn for deadlock", zap.Uint64("txn", se.txn.StartTS()))
 			se.RollbackTxn(ctx)
+			aborted = true
+		}
```

Comment:
I think we can add an `else` on line177, and we return the meetsErr in this `else` statement. Then we can remove `aborted`.

Reply:
But all the branches need to `return meetErr`.

Reply:
But all the branches need to `return meetErr`.

Reply:
PTAL thx